Makefile.e2e

```diff
@@ -189,7 +189,7 @@ test-e2e-kind-mono-repo-test-group3:
 # This target runs all the e2e tests with the multi-repo mode.
 test-e2e-kind-multi-repo: __push-local-image
 	kind delete clusters --all
-	GCP_PROJECT=$(GCP_PROJECT) ./scripts/e2e-kind.sh --multirepo --timeout 60m --parallel 15 --image-tag=$(IMAGE_TAG) $(E2E_ARGS)
+	GCP_PROJECT=$(GCP_PROJECT) ./scripts/e2e-kind.sh --share-test-env --multirepo --timeout 60m --parallel 15 --image-tag=$(IMAGE_TAG) $(E2E_ARGS)
```

Comment:
This is the target used by the presubmit jobs, so my intent here was to make this the way we run presubmits. I haven't ran extensive profiling to measure the diff in runtime. The other (possibly more valuable) benefit is discovering issues with test cleanup before hitting the periodics.

I think [this make target](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile.e2e#L209) probably serves the purpose of local development, although when I'm running these locally I find it a better experience to run these directly using `go test` rather than the make targets. These targets in particular I find pretty unfriendly to local usage since they create a container to run the tests in every time. 

Note the default when running the tests is to not use a shared test env (the `--share-test-env` flag must be provided). I don't feel too strongly about creating various permutations of make targets for the different flags, although at a certain point I feel it's simpler to just have developers run the tests directly with `go test`. Although we should probably do a better job at documenting all of the flag options in our dev guide

Reply:
Can you run some tests to make sure this is actually faster than the old way? I don't want to slow down pre-submits if we don't have to. We can always change this default behavior in a subsequent PR too.

Reply:
I'd love to use this for validating my finalizer PR, but I don't want to make it default without proof that it's not slower. If validating that will take a while, maybe just remove this line to avoid making this the default behavior and move that to another PR.

Reply:
Sure that works for me, I can withhold updating the make target until we get a chance to profile it. You can see a few examples by looking at the current presubmit timings, but that's only a few data points.

From the current executions it seems to be fairly close to the [previous timings](https://oss.gprow.dev/?job=kpt-config-sync-presubmit-e2e-multi-repo-test-group3) (maybe a little faster), but this could probably be improved by optimizing some of the wait conditions during cleanup

Reply:
Removed the change to the make target

/unhold